src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/ExpressionTreeCallRewriter.cs

```diff
@@ -221,7 +217,7 @@ private Expr GenerateLambda(ExprCall pExpr)
         {
             // We always call Lambda(body, arrayinit) where the arrayinit 
             // is the initialization of the parameters.
-            return Visit(pExpr.OptionalArguments.asLIST().OptionalElement);
+            return Visit((pExpr.OptionalArguments as ExprList).OptionalElement);
```

Comment:
I think cases like this where cast result is used to access a member should use a regular ( ) cast.
In unlikely case the cast fails "InvalidCastException" would be easier to diagnose than NRE happening several lines after.